src/TLuaInterpreter.cpp

```diff
@@ -10754,6 +10755,55 @@ bool TLuaInterpreter::compileAndExecuteScript(const QString& code)
     }
 }
 
+// reformats given Lua code. In case of any issues, returns the original code as-is
+// issues could be invalid Lua code or the formatter code bugging out
+QString TLuaInterpreter::formatLuaCode(const QString &code)
+{
+    if (code.isEmpty()) {
+        return code;
+    }
+    lua_State* L = pIndenterState;
+    if (!L) {
+        qDebug() << "LUA CRITICAL ERROR: no suitable Lua execution unit found.";
+        return code;
+    }
+    
+    if (!validLuaCode(code)) {
+        return code;
+    }
+
+    QString escapedCode = code;
+    // escape backslashes so we can pass \n to the function
+    escapedCode.replace(QLatin1String("\\"), QLatin1String("\\\\"), Qt::CaseInsensitive);
+    // escape quotes since we'll be using quotes to pass data to the function
+    escapedCode.replace(QLatin1String("\""), QLatin1String("\\\""), Qt::CaseInsensitive);
+    // escape newlines so they don't interpreted as newlines, but instead get passed onto the function
+    escapedCode.replace(QLatin1String("\n"), QLatin1String("\\n"), Qt::CaseInsensitive);
+
+    QString thing = QString(R"(return get_formatted_code(get_ast("%1"), {indent_chunk = '  ', right_margin = 100, max_text_width = 160, keep_comments = true}))").arg(escapedCode);
+    int error = luaL_dostring(L, thing.toUtf8().constData());
+    QString n;
+    if (error != 0) {
+        string e = "no error message available from Lua";
+        if (lua_isstring(L, 1)) {
+            e = "Lua error:";
+            e += lua_tostring(L, 1);
+        }
+        if (mudlet::debugMode) {
```

Comment:
This is an anti-pattern that should not have been copied IMHO - it means produce `qDebug()` i.e. console debug output ONLY if the Central Debug Console is up on screen; which I think is an obsolete artefact of removed code in the place where it was copied from - and should not be reproduced here. I'd recommend taking the `if(...)` out and leave the `qDebug()` line in unconditionally.

Also, the `e = "Lua error:";` seems a bit questionable to me - it means that the `stdout` output would be of the form:

    LUA ERROR: code did not compile: ERROR: "Lua error: a message from the lua system"

I'd say one or two of the "error"s were surplus there!

Reply:
Just copied the existing error message style. We can adjust the code for all functions in a PR for that specifically!